src/components/select/select.jsx

```diff
@@ -105,7 +105,7 @@ const Select = forwardRef(function Select(props, ref) {
 
   useImperativeHandle(selfRef, () => ({
     get dirty() {
-      return props.value !== this.value;
+      return props.value === undefined ? value !== null : props.value !== value;
```

Comment:
Without this change, `dirty` starts in an incorrect state. It will still look incorrect if `undefined` is intentionally passed as the prop `value`. Not sure how to work around this... collections are so much easier in these cases lol